hummingbot/connector/exchange/kucoin/kucoin_exchange.pyx

```diff
@@ -819,13 +822,20 @@ cdef class KucoinExchange(ExchangeBase):
 
     async def execute_cancel(self, trading_pair: str, order_id: str):
         try:
-            tracked_order = self._in_flight_orders.get(order_id)
+            tracked_order: KucoinInFlightOrder = self._in_flight_orders.get(order_id)
             if tracked_order is None:
                 raise ValueError(f"Failed to cancel order - {order_id}. Order not found.")
+            if tracked_order.is_local:
+                raise KucoinInFlightOrderNotCreated(
+                    f"Failed to cancel order - {order_id}. Order not yet created."
+                    f" This is most likely due to rate-limiting."
+                )
             path_url = f"{CONSTANTS.ORDERS_PATH_URL}/{tracked_order.exchange_order_id}"
             await self._api_request(
                 "delete", path_url=path_url, is_auth_required=True, limit_id=CONSTANTS.DELETE_ORDER_LIMIT_ID
             )
+        except KucoinInFlightOrderNotCreated:
+            raise
```

Comment:
If there is a possibility that the order creation is delayed by the throttler and the cancelation is executed before the creation, maybe we should apply the same fix it was applied for AscendEx, to be sure that the creation is not executed at all afterwards.

Reply:
I forgot to update you on that one @aarmoa: @CrimsonJacket and I had a meeting after our initial one and he pointed out that, on cancel request, we can simply detect that the order is still only "locally" created and not confirmed, and subsequently ignore the request. Eventually, the rate-limited order-creation request will go through, the strategy will request a cancel, and that will go through as well.

It's true that we open ourselves to a possibility where this no-longer wanted order gets submitted post rate-limit and immediately filled, but we reasoned that this is a very unlikely scenario and that the reduction in complexity is worth it.

You can see the code for this in `execute_cancel:828`